base_user_role/models/user.py

```diff
@@ -49,6 +49,11 @@ def write(self, vals):
         self.sudo().set_groups_from_roles()
         return res
 
+    def _get_enabled_roles(self):
+        return self.role_line_ids.filtered(
+            lambda rec: rec.is_enabled and
+            rec.company_id == rec.user_id.company_id)
```

Comment:
In case that the role is not restricted to one specific company we should apply it to all company